### ISSUE TITLE
 0 division results in NaN

### ISSUE THREAD

**Issue by LunaMune** [Prozi/detect-collisions#11]
Function SAT.js @ line 312
```
function circleCircle(a, b, result = null) {
  const a_radius = a.radius * a.scale;
  const b_radius = b.radius * b.scale;
  const difference_x = b.x - a.x; //say this is 0
  const difference_y = b.y - a.y; //say this is 0
  const radius_sum = a_radius + b_radius;
  const length_squared =
    difference_x * difference_x + difference_y * difference_y; //this will be 0

  if (length_squared > radius_sum * radius_sum) {
    return false;
  }

  if (result) {
    const length = Math.sqrt(length_squared); //sqrt 0 == 0

    result.a_in_b = a_radius <= b_radius && length <= b_radius - a_radius;
    result.b_in_a = b_radius <= a_radius && length <= a_radius - b_radius;
    result.overlap = radius_sum - length;
    result.overlap_x = difference_x / length; //division by 0
    result.overlap_y = difference_y / length; // division by 0
  }

  return true;
}
```

**Comment by Prozi** [comment]
@LunaMune 

do you think

```
    result.overlap_x = length ? difference_x / length : 0;
    result.overlap_y = length ? difference_y / length : 0;
```

would solve the issue?

**Comment by usrgnome** [comment]
> @LunaMune
> 
> do you think
> 
> ```
>     result.overlap_x = length ? difference_x / length : 0;
>     result.overlap_y = length ? difference_y / length : 0;
> ```
> 
> would solve the issue?

I believe that it would not.

```
let modifiedLength = length === 0 ? 1 : length; //if length is zero, set the modifiedLength to 1
 result.overlap_x = difference_x / length;
 result.overlap_y = difference_y / length ;
```
In the case of length resulting in 0,  difference_x/y must be zero. This  will create a zero magnitude vector resulting, I believe in the objects remaining in collision?

I suppose this is still a better solution that the positions becoming NaN.

TLDR: Yes, but it will work, but objects will probably remain in collision.

**Comment by Prozi** [comment]
@usrgnome 
@LunaMune 

> ```
> let modifiedLength = length === 0 ? 1 : length; //if length is zero, set the modifiedLength to 1
>  result.overlap_x = difference_x / modifiedLength;
>  result.overlap_y = difference_y / modifiedLength ;
> ```

this would cause

```
    result.overlap = radius_sum - length; // example 100 - 0
    result.overlap_x = difference_x / 1; // still 0
    result.overlap_y = difference_y / 1; // still 0
```

 i have created a fix with tests in such way:

```
    result.overlap_x = difference_x / length || 1;
    result.overlap_y = difference_y / length || 1;
```

now overlap = radius_sum and multiplier is 1

since push back formula is (from readme)

```
object.x -= overlap_x * overlap
object.y -= overlap_y * overlap
```

**Comment by Prozi** [comment]
see https://github.com/Prozi/detect-collisions/commit/c0ac06efd3380246f68ccd046f399e00c983ce24

**Comment by Prozi** [comment]
after version 3 this returns as SAT.Response which has a_in_b and b_in_a and should be handled outside of this library, as you should choose desired direction in case of such collision. will add tests for this in few days/weeks

**Comment by Prozi** [comment]
@usrgnome 
@LunaMune 

added a test just for that case now

https://github.com/Prozi/detect-collisions/blob/master/src/index.spec.js

the case `GIVEN System AND two cricles perfectly overlap THEN they give collision results`

the detect-collisions library correctly detects such collisions,
but they need to be handled by the consuming project

closing the issue and feel free to reopen if you have more comments, or if I misunderstood something
